gen/main.go

```diff
@@ -113,6 +113,38 @@ func loadSchema() (*Schema, error) {
 				}
 			}
 
+			// HACK FreeIPA host has several fields which are not required.
+			hostNotRequiredParams := []string{
+				"subject",
+				"serial_number",
+				"serial_number_hex",
+				"issuer",
+				"valid_not_before",
+				"valid_not_after",
+				"md5_fingerprint",
+				"sha1_fingerprint",
+				"sha256_fingerprint",
+				"managing_host",
+				"ipaallowedtoperform_read_keys_user",
+				"ipaallowedtoperform_read_keys_group",
+				"ipaallowedtoperform_read_keys_host",
+				"ipaallowedtoperform_read_keys_hostgroup",
+				"ipaallowedtoperform_write_keys_user",
+				"ipaallowedtoperform_write_keys_group",
+				"ipaallowedtoperform_write_keys_host",
+				"ipaallowedtoperform_write_keys_hostgroup",
+			}
```

Comment:
I guess you added this workaround since you encountered errors when working with "host" objects. Could you please add a small test which reads (and maybe writes) a host object? That would make it easy to detect similar issues after schema changes in the future.

Reply:
Indeed, these fields failed to parse because they're not present when creating a new host and are required by the schema.